### DISCUSSION TITLE
Content Adressable Storage

### DISCUSSION THREAD

**Discussion by filiphanes** [seaweedfs/seaweedfs#2460] [Ideas]
SeaweedFS curently uses for object ids volumeid + some generated id.
Would it be hard to change fids to hash of content?
Did you thought about it?
I like the idea of volumes. Do you think it could somehow work with content hashes?
I know it could be solved by add another layer of indirection, like storing hashes in filer db, but I would like to avoid lookups.

**Comment by chrislusf** [comment]
Assuming this is the file id definition:
```
<file_id> = <volume_id, needle_id, cookie>
```

The needle_id can be a generated id. But volume_id needs to be flexible, which does not play well with content addressable keys.

Maybe add a bloomfilter layer to check the keys in the volumes.
